src/main/java/com/auth0/client/mgmt/filter/RolesFilter.java

```diff
@@ -0,0 +1,43 @@
+package com.auth0.client.mgmt.filter;
+
+/**
+ * Class used to filter the results received when calling the Grants endpoint. Related to the {@link com.auth0.client.mgmt.RolesEntity} entity.
+ */
+public class RolesFilter extends BaseFilter {
```

Comment:
why reimplementing this? make it extend `PageFilter` and just add "withName"

Reply:
Good call. This has been implemented as suggested.  Please note: I did not adjust any of the pre-existing Filters that also include pagination, as that was outside the scope of this particular PR.

Reply:
Actually, I reverted this. Unless I introduce generics to the Filters, the builder / flow pattern doesn't quite operate the same way. For example, the below would no longer return a RolesFilter:

` RolesFilter filter = new RolesFilter().withTotals(true)`

Reply:
@jakbutler There's no need to use generics. You can override the method on the subclass, call super and make it return "this" just like it's being done [here](https://github.com/auth0/auth0-java/blob/master/src/main/java/com/auth0/client/mgmt/filter/QueryFilter.java#L64-L67). The overriden method will define in it's signature the type being returned, which matches the subclass class.

Reply:
@lbalmaceda - I'd love to see an example; if you attempt a similar change on your local instance, I think you'll understand my response better. I tried what you are describing and using a builder method of a superclass returns an instance of the superclass (not the extending / implementing class), which breaks existing tests unless we use interfaces or generics. On top of that, almost every other Filter class has pagination but does not (nor would, as part of this PR), extend `PageFilter`, so I'm not sure why this is a blocking issue.

Reply:
Not crazy about that approach, but I've implemented it as suggested and pushed.